src/syscheckd/src/db/src/fimDB.cpp

```diff
@@ -199,11 +199,11 @@ void FIMDB::pushMessage(const std::string& data)
 
 void FIMDB::teardown()
 {
-    std::unique_lock<std::shared_timed_mutex> lock(m_handlersMutex);
 
     try
     {
         stopIntegrity();
+        std::lock_guard<std::shared_timed_mutex> lock(m_handlersMutex);
```

Comment:
This is a deadlock detected in development tests.

Reply:
Good catch!